c/Makefile

```diff
@@ -51,18 +53,45 @@ CFLAGS += -std=c99 -O
 # ---- Targets to build ----
 
 LIB = qrcodegen
-LIBFILE = lib$(LIB).a
+ARFILE = lib$(LIB).a
+LIBFILE = lib$(LIB).so
+# Bump the soname number when the ABI changes and gets incompatible
+SO_NAME = $(LIBFILE).1
+REAL_NAME = $(LIBFILE).$(VERSION)
+HEADERS = qrcodegen.h
 LIBOBJ = qrcodegen.o
 MAINS = qrcodegen-demo qrcodegen-test qrcodegen-worker
 
+# define paths to install
+INCLUDEDIR ?= $(DESTDIR)/usr/include/qrcodegen
+LIBDIR ?= $(DESTDIR)/usr/lib
+
 # Build all binaries
-all: $(LIBFILE) $(MAINS)
+all: $(LIBFILE) $(ARFILE) $(MAINS)
 
 # Delete build output
 clean:
-	rm -f -- $(LIBOBJ) $(LIBFILE) $(MAINS:=.o) $(MAINS)
+	rm -f -- $(LIBOBJ) $(LIBFILE) $(ARFILE) $(MAINS:=.o) $(MAINS)
 	rm -rf .deps
 
+install-shared: $(LIBFILE)
+	install -d $(LIBDIR) || true
+	install -m 0644 $(LIBFILE) $(LIBDIR)/$(REAL_NAME)
```

Comment:
Chmod for shared ELF objects must be 0755.

Reply:
That is not true. Libs don't really need to be executable unless you need to execute them they happen to have main function) like i.e. /lib64/libc.so.6 ; which is not the case here.

Reply:
Depends on distribution. On Fedora chmod for all ELF shared objects must be 0755.

I think it will be better to export it as a separate variable:

```
LIBCHMOD ?= 0644
```

And then use:

```
install -m $(LIBCHMOD) $(LIBFILE) $(LIBDIR)/$(REAL_NAME)
```

Reply:
then it makes sense to make it executable for all. But since when it is mandatory on fedora? I am not aware of any such rpmlint check or requirement. The only check we wrote there is if there is the command interface https://github.com/rpm-software-management/rpmlint/pull/286

Reply:
Rpmbuild will not extract debug information from shared ELF objects with chmod != 0755.

Reply:
ouky then, adjusted.